src/Compilers/Test/Core/CompilationVerifier.cs

```diff
@@ -228,6 +233,127 @@ public void VerifyTypeIL(string typeName, string expected)
             }
         }
 
+        private sealed class Resolver : ILVerify.ResolverBase
+        {
+            private readonly Dictionary<string, ImmutableArray<byte>> _imagesByName = new Dictionary<string, ImmutableArray<byte>>(StringComparer.OrdinalIgnoreCase);
+
+            internal Resolver(Dictionary<string, ImmutableArray<byte>> imagesByName)
+            {
+                _imagesByName = imagesByName;
+            }
+
+            protected override PEReader ResolveCore(string simpleName)
+            {
+                if (_imagesByName.TryGetValue(simpleName, out var image))
+                {
+                    return new PEReader(image);
+                }
+
+                throw new Exception($"ILVerify was not able to resolve a module named '{simpleName}'");
+            }
+        }
+
+        private void ILVerify(Verification verification)
+        {
+            if (verification == Verification.Skipped)
+            {
+                return;
+            }
+
+            var imagesByName = new Dictionary<string, ImmutableArray<byte>>(StringComparer.OrdinalIgnoreCase);
+            foreach (var module in _allModuleData)
+            {
+                string name = module.SimpleName;
+                if (imagesByName.ContainsKey(name))
+                {
+                    if ((verification & Verification.FailsILVerify) != 0)
+                    {
+                        return;
+                    }
+
+                    throw new Exception($"Multiple modules named '{name}' were found");
+                }
+                imagesByName.Add(name, module.Image);
+            }
+
+            var resolver = new Resolver(imagesByName);
+            var verifier = new ILVerify.Verifier(resolver);
+            var mscorlibModule = _allModuleData.SingleOrDefault(m => m.IsCorLib);
+            if (mscorlibModule is null &&
+                (verification & Verification.FailsILVerify) != 0)
+            {
+                return;
+            }
+
+            // Main module is the first one
+            var mainModuleReader = resolver.Resolve(_allModuleData[0].SimpleName);
+
+            var (succeeded, errorMessage) = verify(verifier, mscorlibModule.SimpleName, mainModuleReader);
```

Comment:
>mscorlibModule.SimpleName [](http://example.com/codeflow?start=61&length=25)

If `(verification & Verification.FailsILVerify) == 0`, `mscorlibModule` may be null.